metecho/api/tests/jobs.py

```diff
@@ -1231,3 +1235,291 @@ def test_sad(self, scratch_org_factory, user_factory):
+        and the repository is created successfully
+        """
+        project = project_factory(github_users=({"login": "user1"}, {"login": "user2"}))
+
+        team = mocker.MagicMock()
+        gh_user = mocker.patch(f"{PATCH_ROOT}.gh_given_user").return_value
+        gh_user.organizations.return_value = [
+            mocker.MagicMock(login=project.repo_owner),
+        ]
+        gh_org = mocker.patch(f"{PATCH_ROOT}.get_org_for_repo_creation").return_value
+        gh_org.create_team.return_value = team
+        gh_org.create_repository.return_value = mocker.MagicMock(
+            id=123456, html_url="", permissions=[]
+        )
+
+        return project, team
+
+    def test_ok(self, mocker, github_mocks, user_factory):
+        user = user_factory()
+        project, team = github_mocks
+        mocker.patch(f"{PATCH_ROOT}.init_from_context")
+        sarge = mocker.patch(f"{PATCH_ROOT}.sarge", autospec=True)
+        sarge.capture_both.return_value.returncode = 0
+        async_to_sync = mocker.patch("metecho.api.model_mixins.async_to_sync")
+
+        create_repository(project, user=user, dependencies=["http://foo.com"])
+        project.refresh_from_db()
+
+        assert project.repo_id == 123456
+        assert user.repositories.filter(repo_id=123456).exists()
+        assert (
+            team.add_or_update_membership.call_count == 3
+        ), "Expected three calls: one for the user and two for the collaborators"
+        async_to_sync.return_value.assert_called_with(
+            project,
+            {
+                "type": "PROJECT_UPDATE",
+                "payload": {"originating_user_id": str(user.pk)},
+            },
+            for_list=False,
+            group_name=None,
+        )
+        assert sarge.capture_both.called
+
+    def test__gh_error(self, mocker, caplog, project, user_factory):
+        user = user_factory()
+        async_to_sync = mocker.patch("metecho.api.model_mixins.async_to_sync")
+        mocker.patch(f"{PATCH_ROOT}.gh_given_user", side_effect=Exception("Oh no!"))
+
+        with pytest.raises(Exception, match="Oh no!"):
+            create_repository(project, user=user, dependencies=[])
+
+        with pytest.raises(project.DoesNotExist):
+            # Expect project to be deleted from the DB since repo creation failed
+            project.refresh_from_db()
+
+        assert "Oh no!" in caplog.text
+        async_to_sync.return_value.assert_called_with(
+            project,
+            {
+                "type": "PROJECT_UPDATE_ERROR",
+                "payload": {
+                    "originating_user_id": str(user.pk),
+                    "message": "Oh no!",
+                },
+            },
+            for_list=False,
+            group_name=None,
+        )
+
+    def test__push_error(self, mocker, caplog, github_mocks, user_factory):
+        user = user_factory()
+        mocker.patch(f"{PATCH_ROOT}.init_from_context")
```

Comment:
@davisagli for some reason the tests fail if I don't patch the new `init_from_context` function. It appears to be an error with Jinja that only happens during test runs. Have you encountered something like that before?

```
metecho/api/tests/jobs.py:1328: in test__push_error
    create_repository(project, user=user, dependencies=[])
metecho/api/jobs.py:258: in create_repository
    init_from_context(context)
/usr/local/lib/python3.9/site-packages/cumulusci/cli/project.py:228: in init_from_context
    template = env.get_template(name)
/usr/local/lib/python3.9/site-packages/jinja2/environment.py:883: in get_template
    return self._load_template(name, self.make_globals(globals))
/usr/local/lib/python3.9/site-packages/jinja2/environment.py:857: in _load_template
    template = self.loader.load(self, name, globals)
/usr/local/lib/python3.9/site-packages/jinja2/loaders.py:111: in load
    source, filename, uptodate = self.get_source(environment, name)
/usr/local/lib/python3.9/site-packages/jinja2/loaders.py:242: in get_source
    if not self.provider.has_resource(p):
/usr/local/lib/python3.9/site-packages/pkg_resources/__init__.py:1391: in has_resource
    return self._has(self._fn(self.module_path, resource_name))
/usr/local/lib/python3.9/site-packages/pkg_resources/__init__.py:1458: in _has
    raise NotImplementedError(
E   NotImplementedError: Can't perform this operation for unregistered loader type
```